### DISCUSSION TITLE
sending arguments into data loaders using an interactive user-driven rig

### DISCUSSION THREAD

**Discussion by drewda** [observablehq/framework#1743] [Q&A] [answered]
We've been making great use of Observable Framework to create reports on a recurring batch basis.

We're now thinking about whether we might be able to also use Framework in a more interactive batch architecture.

Our goal would be for users to be able to provide some arguments for a report. We'd probably build our own basic web UI for this. After the user submits the arguments, an async background job would run Framework in a sandboxed environment, publish the output, and direct the user to the static assets.

If any of the contributors are up for considering this: Is there an existing way to send arguments into Framework data loaders? Looks like https://github.com/observablehq/framework/issues/245 has added some potentially related capabilities.

Also open to any other input or ideas. Thank you!

**Comment by mythmon** [comment, marked as answer]
That's a very interesting idea! I'd probably approach this by setting environment variables when running the Framework build command. Those environment variables could then be read in the data loaders.

Alternatively, the `observablehq.config.js` file is code, and is free to run arbitrary code. It could read the user input from some input (maybe environment variables, maybe a file, maybe something else?). Then it could construct `dynamicPaths` in such a way to trigger the parameterized data loaders feature from #245.

I'm curious to hear about what you build with this!

**Reply by drewda** [reply]
Thanks for such a prompt response and for these pointers!
